### ISSUE TITLE
Xtext should ignore files in Gradle 'build' folders

### ISSUE THREAD

**Issue by meysholdt** [eclipse/xtext-eclipse#36]
Buildship seems to hide the 'build' folder per default in the Eclipse Package Explorer.
The Xtext incremental Builder, however, seems to process files from there and potentially creates validation error markers. This is very confusing for users, because the markers reside on files that are not visible in the Package Explorer.

**Comment by meysholdt** [comment]
IIRC we have/had the same problem with Maven's 'target' folder.

**Comment by meysholdt** [comment]
What's the best way for Xtext to determine which derived files should be build and which ones should not be build?

**Comment by cdietrich** [comment]
from @oehme 

> Use resource filters in your project (Gradle will soon provide an API to define those, so this could be encapsulated in a Gradle plugin)

=> we should wait for this

**Comment by cdietrich** [comment]
@oehme any news on this api?

**Comment by oehme** [comment]
No, the contributor did not follow up with the Buildship part that would use that API.

But I don't think that resource filters are the answer here anyway. Why are there models in the build folder?

**Comment by cdietrich** [comment]
cause they are copied to the resources/test resources folders

**Comment by oehme** [comment]
And the reason we don't ignore the whole folder is because there are some users who generate code that generates code?

**Comment by cdietrich** [comment]
i am not sure about that. that isnt working in xtext anyway without a custom builder that touches files cause a builder wont get notified on changes it produced itself.

**Comment by szarnekow** [comment]
I think the problem is that the Eclipse Project model does not support multiple output folders per source folder and we have the model files in the same source folder as the java files. So build/resources/test (?) is not marked as output folder and thereby not automatically ignored.

**Comment by szarnekow** [comment]
What we could try is to mark the entire build folder as output folder for the project and than use the 'real' output folder on a source folder level?

**Comment by cdietrich** [comment]
actually we even use bin as output folder .....

**Comment by oehme** [comment]
The build folder is marked as derived by Buildship, so if Xtext ignored derived resources it would be fine I think. But that may break some edge cases I guess?

**Comment by cdietrich** [comment]
yes it will break the secario described above and additional scenarios where xtext file are generated by another builder
